server/error.go

```diff
@@ -166,11 +167,12 @@ func NewInvocationCapabilityError(capabilities []ucan.Capability[any]) Invocatio
 }
 
 type InvalidAudienceError struct {
-	expected, actual string
+	accepted []string
+	actual   string
 }
 
 func (i InvalidAudienceError) Error() string {
-	return fmt.Sprintf("Invalid audience: expected %s, got %s", i.expected, i.actual)
+	return fmt.Sprintf("Invalid audience: accepted %s, got %s", strings.Join(i.accepted, ", "), i.actual)
```

Comment:
could still be "expected" if you join with `" or "`

Reply:
yeah, that works too. I reverted back to `expected` and also simplified using the error by accepting `ucan.Principals` directly and doing the stringification in the error type itself.